flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -223,17 +228,42 @@ public void run() {
 							}
 						}
 					}
-					if (updateOverview) {
-						updateJobOverview(webOverviewDir, webDir);
-						for (int x = 0; x < numFetchedArchives; x++) {
-							numArchivedJobs.countDown();
-						}
-					}
 				}
+
+				if (!jobsToRemove.isEmpty()) {
+					operations.addAll(removeDeletedJobs(jobsToRemove));
+				}
+				if (!operations.isEmpty()) {
+					updateJobOverview(webOverviewDir, webDir);
+				}
+				operations.forEach(operation ->
+					jobArchiveOperationListener.accept(operation.getLeft(), operation.getRight()));
 			} catch (Exception e) {
 				LOG.error("Critical failure while fetching/processing job archives.", e);
 			}
 		}
+
+		private List<Pair<JobArchiveOperation, String>> removeDeletedJobs(Set<String> jobsToRemove) {
+
+			List<Pair<JobArchiveOperation, String>> deleteOperations = new ArrayList<>();
+			if (LOG.isInfoEnabled()) {
```

Comment:
this is unnecessary as toString() on jobsToRemove will be only called in case info level is enabled

Reply:
Thank you, removed